Overall improvements

peduarte opened the pull request:
This PR:
- Set up blog structure
- New demos for all component
- Updated code blocks
- Automatically create code block based on a file (used for demos)
- Automatically create CodeSanboxes based on the demo
- Move Label and Aspect Ratio to Components
- Set up redirects for Label and Asect Ratio

benoitgrelard commented:
Brilliant work 👏 
Will take a look later and go through all the demos see if anything jumps out.

benoitgrelard commented on components/demos/Toolbar.jsx:
```diff
@@ -0,0 +1,126 @@
+  lineHeight: 1,
+  alignItems: 'center',
+  justifyContent: 'center',
+  '&:hover': { backgroundColor: violet.violet4, color: violet.violet11 },
```
not sure if it is but the hover background seems darker than on the single toggle demo, and so it's not a big change between both and I sometimes confused whether it's selected or my cursor is over it. Also worse on mobile because hover sticks so it really looks selected.

benoitgrelard commented on data/primitives/components/accordion/0.0.17.mdx:
```diff
@@ -32,12 +38,12 @@ npm install @radix-ui/react-accordion
 ## Anatomy
 
-Import the components and piece the parts together.
+Import all parts and piece the parts together.
```
```suggestion
Import all parts and piece them together.
```

benoitgrelard replied:
Probably worth updating this sentence in each of the latest version for each primitives

benoitgrelard reviewed (changes requested):
@peduarte this is just AMAZING! Straight 🔥🔥🔥

I've gone through all the pages and demos and it's overall amazing stuff.
I've looked out for accessibility and have added comments mostly about missing labelling in the demos plus a few odds and ends here and there.

I've also checked all the demos on mobile and checked every single codesandbox export to see if they looked the same, found a few things there too.

I haven't gone too much through the files cause there was so much and I'm not sure what you actually changed vs prettier, etc.

Apart from that, a few general comments:
- is it worth revamping the "getting started" demo to match the rest? Possibly even reuse the one we already have created?
- default `-webkit-tap-highlight-color` on mobile is kinda annoying
- anything of the demos that has a component that uses the `RemoveScroll` doesn't fair well with that body top padding you have. It makes the whole thing jump when open. We should probably add the padding somewhere else. Here's an example: 
![image](https://user-images.githubusercontent.com/1539897/125692832-4da8c335-f0a8-4f3c-8d69-13b8d3e27a99.png)

peduarte commented on components/demos/AspectRatio.jsx:
```diff
@@ -0,0 +1,36 @@
+  objectFit: 'cover',
+  width: '100%',
+  height: '100%',
+  objectPosition: 'bottom',
```
It was an Easter egg 😅 chose that photo for the gradient haha looks so sick

peduarte commented on data/primitives/components/aspect-ratio/0.0.13.mdx:
```diff
@@ -80,11 +66,7 @@ Contains the content you want to constrain to a given ratio.
 
 ### 16 x 9 ratio
```
Let's remove it

peduarte commented on components/demos/Label.jsx:
```diff
@@ -0,0 +1,44 @@
+    <Label htmlFor="firstName" css={{ lineHeight: '35px', marginRight: 15 }}>
+      First name
+    </Label>
+    <Input type="text" id="firstName" value="Pedro Duarte" />
```
That was on purpose

peduarte commented on components/demos/Slider.jsx:
```diff
@@ -0,0 +1,63 @@
+  boxShadow: `0 2px 10px ${blackA.blackA7}`,
+  borderRadius: 10,
+  '&:hover': { backgroundColor: violet.violet3 },
+  '&:focus': { boxShadow: `0 0 0 5px ${blackA.blackA7}` },
```
Yeah, it is. Works better in this case. Could make it a notch darker though

peduarte commented:
@benoitgrelard Thank you! 🚀  Resolved all comments.

> is it worth revamping the "getting started" demo to match the rest? Possibly even reuse the one we already have created?
Done

> anything of the demos that has a component that uses the RemoveScroll doesn't fair well with that body top padding you have. It makes the whole thing jump when open. We should probably add the padding somewhere else. Here's an example:
Sorted

benoitgrelard commented:
The codesandbox export for the demo in the getting started doesn't work because it doesn’t have the dep name.
It doesn’t know it’s popover.
I think cause you normally rely on the page to know that right?

![image](https://user-images.githubusercontent.com/1539897/125753544-e9c77a69-2b6b-4b5a-9125-6ca39acbae17.png)

`@radix-ui/react-undefined`

peduarte commented:
> The codesandbox export for the demo in the getting started doesn't work because it doesn’t have the dep name.

Makes sense 👌 I've made some changes. Treating this as a normal codeblock with preview, instead of a "Hero".

benoitgrelard reviewed (approved):
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

andy-hook commented:
Amazing work @peduarte this looks fantastic! such a great improvement. A couple of tiny things that stood out to me:

![image](https://user-images.githubusercontent.com/11708259/125760016-30ad0c1f-5fef-4ec4-baf8-b46c54bd2b18.png)

Nothing against the moon but my very first reaction was to focus on the shape and it kind of resembles a loading indicator. Perhaps an image with full bleed details might better illustrate the aspect / cropping utility of this prim?

![image](https://user-images.githubusercontent.com/11708259/125760128-ca958a3b-67bd-4b46-aab2-3b4bb43b465a.png)

We make a point in the first bullet about the delay option and I was wondering whether it's worth having multiple elements to show that in action?

peduarte commented:
@andy-hook Cheers man <3

Good shout about the moon haha, I've replaced the image now, let me know what you think?

![CleanShot 2021-07-15 at 14 29 44@2x](https://user-images.githubusercontent.com/372831/125788545-b51abbde-5e70-4e62-a8ad-f84a48cf80c8.jpg)


About the tooltip: good point. I think for the sake of simplicity we should have just one tooltip in the hero, but I've updated the "Show instantly" example so people know how to that if they need to

Cheers! 🙏

benoitgrelard commented on components/demos/AspectRatio.jsx:
```diff
@@ -12,7 +12,7 @@ const Img = styled('img', {
   width: '100%',
   height: '100%',
-  objectPosition: 'center 30%',
+  objectPosition: 'center center',
```
Pretty sure that's the default, so we can drop this line alltogether.

andy-hook commented:
> let me know what you think?

Looks 🔥 

> I think for the sake of simplicity we should have just one tooltip in the hero

No worries, sounds good 👍

colmtuite commented:
Excellent work 👏
